storage: deflake TestLeaseExtensionNotBlockedByRead, TestRemoteDebugModeSetting

tbg opened the pull request:
We're waiting less than we used to before going into the meat of the
test, so now our lease request can catch a reproposal due to the leader
election, which the test has to handle gracefully. Luckily that
particular lease request is idempotent, so we just retry.

Fixes #33579.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33585)
<!-- Reviewable:end -->

tbg committed:
storage: deflake TestLeaseExtensionNotBlockedByRead

We're waiting less than we used to before going into the meat of the
test, so now our lease request can catch a reproposal due to the leader
election, which the test has to handle gracefully. This is awkward and
ideally we wouldn't have these maybe-faux reproposals in the first
place, but they are often necessary (until we use Raft's
ErrProposalDropped). PS: trying to settle the elections is more awkward
as far as I'm concerned.

Fixes #33579.

Release note: None

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33585#-:-LVs9k4X-zzgmLcsS73x:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/server/status.go, line 280 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/33585#-LVs9airEFAwRyLG1HSF:-LVs9airEFAwRyLG1HSG:b2xl99j) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e22cd9d1f2bd79db76b8c8fe9d672751e67b7a5/pkg/server/status.go#L280)):*
> ```Go
> 					rep, err := store.GetReplica(desc.RangeID)
> 					if err != nil {
> 						return true, nil // continue
> ```

What error is being ignored here? Perhaps add a comment about why swallowing the error is the right thing to do.


<!-- Sent from Reviewable.io -->

tbg committed:
server: don't bail when a range is missing

Make the allocator endpoint more resilient to replicas moving
around or getting merged.

Fixes #33496.

Release note: None

tbg reviewed (commented):
TFTR!

bors r=petermattis

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33585#-:-LVsE6z28_vakPejW5c1:b-spmsx3)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/server/status.go, line 280 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/33585#-LVs9airEFAwRyLG1HSF:-LVsDz0L7wWpepobRrLK:b-uhler5) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e22cd9d1f2bd79db76b8c8fe9d672751e67b7a5/pkg/server/status.go#L280)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

What error is being ignored here? Perhaps add a comment about why swallowing the error is the right thing to do.
</blockquote></details>

It's always RangeNotFound, but I updated the code to be robust.


<!-- Sent from Reviewable.io -->
